objects/cash_register.rb

```diff
@@ -0,0 +1,32 @@
+#! /usr/bin/env ruby
+
+class CashRegister
+  def initialize
+    @total = 0.0
+  end
+
+  def purchase(amount)
+    @total += amount
+  end
+
+  def format_total
+    sprintf("$%.2f", @total.abs)
+  end
+
+  def total
+    puts format_total
```

Comment:
This method should just be keeping track of the total. Also, if you added an attr_reader to the class for total, the place in the code where you are only reading the total (not setting it) `@total` can be changed to just `total`